src/akar/util.clj

```diff
@@ -0,0 +1,42 @@
+(ns akar.util
+  (:require [clojure.walk :as cw]))
+
+(defn postwalk-fn-replace
+  "Calls the matching-fn on every expression and if matching-fn
+  returned true, replaces the expression by the result of calling
+  replacing-fn with the given expr as the parameter"
+  [matching-fn replacing-fn form]
+  (cw/postwalk (fn [x]
+                   (if (matching-fn x)
+                     (replacing-fn x)
+                     x))
+                 form))
+
+
+(defmacro def-trampolined-recur
+  "For recursive functions using recur along with akar match clauses,
+  this macro generates a tail call optimized function body using
+  trampoline.
+
+  This generates two functions,
+  1. Function with \"*\" suffixed to the name give. The body for
+  this function is same as the body given except the recur call replaced
+  with an anonymous function calling the same function recursively.
+
+  2. A function with the name - same as the name given in parameter. This
+  function calls the above generated function using trampoline.
+
+  This is done to avail the tail call optimization along with using the akar
+  patterns"
+  [nm args & body]
+  (let [tail-rec-fn-name (symbol (str nm "*"))
+        fn-body# (postwalk-fn-replace (fn [expr]
+                                        (and (seq? expr)
+                                             (= (first expr) 'recur)))
+                                      (fn [expr]
+                                        `(fn [] ~(cons tail-rec-fn-name (rest expr))))
+                                      (cons 'do body))
+        inner-fn# `(defn ~tail-rec-fn-name ~(vec args) ~fn-body#)
+        tramp-fn# `(defn ~nm ~(vec args) (trampoline ~tail-rec-fn-name ~@(vec args)))]
```

Comment:
`trampoline` does not require a bound var, but a plain function. You could pass an anonymous function here. 

[Here](https://stackoverflow.com/a/10490737/192247) is how to recur in a Clojure anonymous function.

Reply:
yeah, this is cool.